core/domain/platform_parameter_domain.py

```diff
@@ -397,22 +404,60 @@ def _is_first_version_smaller(self, version_a, version_b):
                 return False
         return False
 
-    def _match_version_flavor(self, flavor, client_version):
+    def _match_version_flavor(self, op, flavor, client_version):
         """Matches the client version flavor.
 
         Args:
+            op: str. The operator for comparison, e.g. '=', '>'.
             flavor: str. The flavor to match, e.g. 'alpha', 'beta', 'test',
                 'n/a'.
             client_version: str. The version of the client, given in the form
                 of '<version>-<hash>-<flavor>'. The hash and flavor of client
                 version is optional, if absent, the flavor is considered 'n/a'.
 
         Returns:
-            bool. True is the client_version matches the given flavor.
+            bool. True is the client_version matches the given flavor using
+            the operator.
         """
         match = APP_VERSION_WITH_HASH_REGEXP.match(client_version)
-        client_flavor = match.group(2) or 'unknown'
-        return client_flavor == flavor
+        client_flavor = match.group(2)
+
+        if client_flavor is None:
+            return False
+
+        is_equal = flavor == client_flavor
+        is_client_flavor_smaller = self._is_first_flavor_smaller(
+            client_flavor, flavor)
+        is_client_flavor_larger = self._is_first_flavor_smaller(
+            flavor, client_flavor)
+
+        if op == '=':
+            return is_equal
+        elif op == '<':
+            return is_client_flavor_smaller
+        elif op == '<=':
+            return is_equal or is_client_flavor_smaller
+        elif op == '>':
+            return is_client_flavor_larger
+        elif op == '>=':
+            return is_equal or is_client_flavor_larger
+
+    def _is_first_flavor_smaller(self, flavor_a, flavor_b):
+        """Compares two version flavors that are not 'unknown', return True
+        if the first version is smaller in the following ordering:
+        'alpha' < 'beta' < 'test' < 'release'
+
+        Args:
+            flavor_a: str. The version flavor.
+            flavor_b: str. The version flavor.
+
+        Returns:
+            bool. True if the first flavor is smaller.
+        """
+        return (
+            ALLOWED_APP_VERSION_FLAVOR.index(flavor_a) <
```

Comment:
Please update the documentation for ``ALLOWED_APP_VERSION_FLAVOR`` to make it clear that its ordering matters & how.

Reply:
Done, added description next to the definition of `ALLOWED_APP_VERSION_FLAVOR`.